salalib/mgraph.cpp

```diff
@@ -1312,44 +1314,44 @@ int MetaGraph::loadLineData( Communicator *communicator, int load_type )
          map.open(communicator->GetFileSet(), communicator);
       }
       catch (Communicator::CancelledException) {
-         SuperSpacePixel::pop_back();
+         m_spacePixels.pop_back();
          return 0;
       }
       catch (pexception) {
-         SuperSpacePixel::pop_back();
+         m_spacePixels.pop_back();
          return -1;
       }
 
       if (communicator->IsCancelled()) {
-         SuperSpacePixel::pop_back();
+         m_spacePixels.pop_back();
          return 0;
       }
 
-      SuperSpacePixel::tail().m_region = map.getRegion();;
+      m_spacePixels.back().m_region = map.getRegion();;
 
       for (size_t i = 0; i < map.size(); i++) {
 
-         SuperSpacePixel::tail().push_back(ShapeMap(map[i].getName()));
-         SuperSpacePixel::tail().at(i).init(map[i].getLineCount(), map.getRegion());
+         m_spacePixels.back().m_spacePixels.emplace_back(map[i].getName());
+         m_spacePixels.back().m_spacePixels[i].init(map[i].getLineCount(), map.getRegion());
 
          for (size_t j = 0; j < map[i].size(); j++) {
 
             for (size_t k = 0; k < map[i][j].size(); k++) {
 
-               SuperSpacePixel::tail().at(i).makeLineShape( map[i][j][k] );
+               m_spacePixels.back().m_spacePixels[i].makeLineShape( map[i][j][k] );
             }
          }
 
-         SuperSpacePixel::tail().at(i).setDisplayedAttribute(-2);
-         SuperSpacePixel::tail().at(i).setDisplayedAttribute(-1);
+         m_spacePixels.back().m_spacePixels[i].setDisplayedAttribute(-2);

```

Comment:
I think I know why the displayed attribute needs to be set to -2 before being set to -1, but it's still a code smell of "this fish hasn't see the sea for at least 2 weeks" magnitude...

Reply:
I really wish I didn't know.